server/src/tokens.rs

```diff
@@ -118,6 +119,15 @@ impl FromRequest for EdgeToken {
             None => "Authorization".to_string(),
         };
         let value = req.headers().get(token_header);
+
+        if let (Some(value), Some(token_cache)) =
```

Comment:
The other piece of magic here - this function materializes a token from a request- (including all the app scoped data. If that app scoped data has a token cache and that token cache has this token stored then skip the parsing and return that value

Reply:
Because we only keep tokens that are valid in our token cache, right? I know there's no such thing for these new trusted tokens (once supplied, they're always valid), but I assume that if we have an Unleash token that becomes invalid we remove it from the cache. Which means we can trust our cache: If the token is there, it's valid.

Reply:
We actually keep around invalid tokens too (i.e. tokens that we've asked Unleash about and Unleash has actively said it's not valid), mostly so we can short circuit the Forbidden instead of trying to re-validate them. In practice I don't think that'd be a problem here since we never query these against Unleash itself, I'm just trying to scope this as narrowly as possible so I can reason about the different validation flows

The logic for ingesting and validating tokens needs a look over because it's way too complex for what it's doing but I don't want to touch that in this PR

Reply:
Got it, makes sense 👍 

Also, I guess there will be no way of revoking a trusted token unless you remove it from your params and restart your Edge instance, but that's probably a compromise we're okay with. Maybe it should be documented as a pro/con of Unleash tokens vs trusted tokens.

Reply:
Yuh, works like the old trusted tokens on the Proxy. Requires a hard restart and parameter change to revoke them. Feels like an okay compromise

Yessir! This definitely needs some docs